lib/puppet/provider/iis_httperror/iis_httperror.rb

```diff
@@ -0,0 +1,177 @@
+require 'json'
+
+Puppet::Type.type(:iis_httperror).provide(:iis_httperror) do
+	confine :operatingsystem => :windows
+	defaultfor :operatingsystem => :windows
+
+	commands :powershell =>
```

Comment:
You don't have to setup a powershell command yourself, if you inherit the `iis_powershell` class we use the new PowerShellManager code that optimizes calls to PowerShell. It also handles all the JSON conversion and handling PowerShell v2 semantics.